Amplify/Categories/DataStore/CustomCodable/CustomCodable.swift

```diff
@@ -0,0 +1,17 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+// MARK: - CustomCodable
+
+/// All persistent custom types should conform to the CustomCodable protocol.
+public protocol CustomCodable: Codable {
```

Comment:
I think we need a better name for `CustomCodable`. ORMs typically call these *"Embedded"* types, maybe we could follow that?

@jpignata also mentioned the *non-model* thing we keep talking about is not clear, and I agree.